conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala

```diff
@@ -0,0 +1,136 @@
+import tech.cryptonomic.conseil.common.ethereum.rpc.json.{Block, Log, Transaction}
+import tech.cryptonomic.conseil.common.ethereum.EthereumPersistence._
+
+/**
+  * Ethereum persistence into the database using Slick.
+  */
+class EthereumPersistence[F[_]: Concurrent] extends LazyLogging {
+
+  /**
+    * Create [[DBIO]] seq with blocks and transactions that can be wrap into one transaction.
+    *
+    * @param block JSON_RPC block
+    * @param transactions JSON_RPC block's transactions
+    */
+  def createBlock(
+      block: Block,
+      transactions: List[Transaction]
+  ): DBIOAction[Unit, NoStream, Effect.Write] =
+    DBIO.seq(
+      Tables.Blocks += block.convertTo[Tables.BlocksRow],
+      Tables.Transactions ++= transactions.map(_.convertTo[Tables.TransactionsRow])
+    )
+
+  /**
+    * Get sequence of existing blocks heights from the database.
+    *
+    * @param range Inclusive range of the block's height
+    */
+  def getIndexedBlockHeights(range: Range.Inclusive): DBIO[Seq[Int]] =
+    Tables.Blocks.filter(_.number between (range.start, range.end)).map(_.number).result
+
+  /**
+    * Get the latest block from the database.
+    */
+  def getLatestIndexedBlock: DBIO[Option[Tables.BlocksRow]] =
+    Tables.Blocks.sortBy(_.number.desc).take(1).result.headOption
+}
+
+object EthereumPersistence {
+
+  /**
+    * Create [[cats.Resource]] with [[EthereumPersistence]].
+    */
+  def resource[F[_]: Concurrent]: Resource[F, EthereumPersistence[F]] =
+    Resource.pure(new EthereumPersistence[F])
+
+  /**
+    * Convert form [[Block]] to [[Tables.BlocksRow]]
+    * TODO: This conversion should be done with the Chimney,
+    *       but it's blocked due to the https://github.com/scala/bug/issues/11157
+    */
+  implicit val blockToBlocksRow: Conversion[Id, Block, Tables.BlocksRow] = new Conversion[Id, Block, Tables.BlocksRow] {
+    override def convert(from: Block) =
+      Tables.BlocksRow(
+        hash = from.hash,
+        number = Integer.decode(from.number),
+        difficulty = from.difficulty,
+        extraData = from.extraData,
+        gasLimit = from.gasLimit,
+        gasUsed = from.gasUsed,
+        logsBloom = from.logsBloom,
+        miner = from.miner,
+        mixHash = from.mixHash,
+        nonce = from.nonce,
+        parentHash = from.parentHash,
+        receiptsRoot = from.receiptsRoot,
+        sha3Uncles = from.sha3Uncles,
+        size = from.size,
+        stateRoot = from.stateRoot,
+        totalDifficulty = from.totalDifficulty,
+        transactionsRoot = from.transactionsRoot,
+        uncles = Option(from.uncles).filter(_.nonEmpty).map(_.mkString(",")),
+        timestamp = Timestamp.from(Instant.ofEpochSecond(Integer.decode(from.timestamp).toLong))
+      )
+  }
+
+  /**
+    * Convert form [[Transaction]] to [[Tables.TransactionsRow]]
+    * TODO: This conversion should be done with the Chimney,
+    *       but it's blocked due to the https://github.com/scala/bug/issues/11157
```

Comment:
I've tested on this case and chimney transformation compiles correctly:
```
      import io.scalaland.chimney.dsl._

      override def convert(from: Transaction) =
        from
          .into[Tables.TransactionsRow]
          .withFieldComputed(_.blockNumber, x => Integer.decode(x.blockNumber).toInt)
          .transform
```
Can it be replaced in this case, or am I missing something?

Reply:
shouldn't it be done for all places with the `tech.cryptonomic.conseil.common.util.Conversion`? I think that we should get rid of `Conversion` first, then refactor all the places with it to use only `chimney`. Of course, I can only change that here too. @ivanopagano WDYT?

Reply:
I'll give you some feedback on this, but won't be able right now.

I need to test the effect on updating to latest chimney (as suggested to me) to see if it solves anything.

I've sent a bugfix for that issue mentioned, but it doesn't seem to solve the issue on my side when I tried locally... so it might not be related.

Eventually, the latest chimney api would allow us to get rid of the Conversions, I think. And be able to operate directly with the chimney-defined type-class which does the same job...